src/sentry/tasks/unmerge.py

```diff
@@ -0,0 +1,468 @@
+    'score': lambda data, event: ScoreClause.calculate(
+        data['times_seen'] + 1,
+        data['last_seen'],
+    ),
+}
+
+
+def get_group_creation_attributes(events):
+    latest_event = events[0]
+    return reduce(
+        lambda data, event: merge_mappings([
+            data,
+            {name: f(data, event) for name, f in backfill_fields.items()},
+        ]),
+        events,
+        {name: f(latest_event) for name, f in initial_fields.items()},
+    )
+
+
+def get_group_backfill_attributes(group, events):
+    return {
+        k: v for k, v in
+        reduce(
+            lambda data, event: merge_mappings([
+                data,
+                {name: f(data, event) for name, f in backfill_fields.items()},
+            ]),
+            events,
+            {name: getattr(group, name) for name in set(initial_fields.keys()) | set(backfill_fields.keys())},
+        ).items()
+        if k in backfill_fields
+    }
+
+
+def get_fingerprint(event):
+    # TODO: This *might* need to be protected from an IndexError?
+    primary_hash = get_hashes_for_event(event)[0]
+    return md5_from_hash(primary_hash)
+
+
+def migrate_events(project, destination_id, fingerprints, events):
+    # XXX: This is only actually able to create a destination group and migrate
+    # the group hashes if there are events that can be migrated. How do we
+    # handle this if there aren't any events? We can't create a group (there
+    # isn't any data to derive the aggregates from), so we'd have to mark the
+    # hash as in limbo somehow...?)
+    if not events:
+        return destination_id
+
+    if destination_id is None:
+        # XXX: There is a race condition here between the (wall clock) time
+        # that the migration is started by the user and when we actually
+        # get to this block where the new destination is created and we've
+        # moved the ``GroupHash`` so that events start being associated
+        # with it. During this gap, there could have been additional events
+        # ingested, and if we want to handle this, we'd need to record the
+        # highest event ID we've seen at the beginning of the migration,
+        # then scan all events greater than that ID and migrate the ones
+        # where necessary. (This still isn't even guaranteed to catch all
+        # of the events due to processing latency, but it's a better shot.)
+        # Create a new destination group.
+        destination = Group.objects.create(
+            project_id=project.id,
+            short_id=project.next_short_id(),
+            **get_group_creation_attributes(events)
+        )
+
+        destination_id = destination.id
+
+        # Move the group hashes to the destination.
+        # TODO: What happens if this ``GroupHash`` has already been
+        # migrated somewhere else? Right now, this just assumes we have
+        # exclusive access to it (which is not a safe assumption.)
```

Comment:
Was thinking we'd probably add a status column to `GroupHash` to allow best-effort exclusive access here.